### DISCUSSION TITLE
Unable to assign VLAN to prefix, if the prefix is assigned to a site

### DISCUSSION THREAD

**Discussion by jmmey** [netbox-community/netbox#6911] [Q&A]
# Problem description

If a prefix has a site assignment, it is unable to be assigned a VLAN. The only work around I found is to remove the site assignment from the prefix. At that point the UI will allow me to assign a VLAN to the prefix. 

Has anyone else experience this issue in version `2.11.10`? This is happening on two different installations of netbox. I'm not sure if it is something I do not understand from the docs or a bug.

## Steps to Recreate

1. Create a VLAN with an ID, Name, and Status set to "Active".
2. Create a site.
3. Create an IPv4 prefix, and assign it to the site created in step 2, save.
4. Edit the IPv4 prefix from step 3, under section _Site/VLAN Assignment_, select the drop down for VLAN. The drop down states "No results found".

<img width="729" alt="no-vlan" src="https://user-images.githubusercontent.com/46431837/128588106-d13e5b05-dfc4-4d75-827d-6e7abc921133.png">

5. Still in the IPv4 prefix edit screen under section _Site/VLAN Assignment_, remove the site by either selecting "None" or click the `x`.
6. Click on the drop down for VLAN, and now the list is populated with a list of VLANs available for assignment.

<img width="714" alt="vlan" src="https://user-images.githubusercontent.com/46431837/128588137-a07758f1-8a85-4a45-904e-4fadc41e8a2d.png">

**Comment by candlerb** [comment]
Have you tried:
* assigning the VLAN to a site?
* assigning the VLAN to a VLANGroup, and assigning the VLANGroup to a site?

**Reply by jmmey** [reply]
# Here are the results from those two suggestions.

## Assigning the VLAN to a site

- VLAN assigned to the same site as the prefix works, but the VLAN can no longer be assigned to a VLAN group.

URL for image: `https://netbox-url/ipam/vlans/1/edit/`

![vlan-group-site-scope-error](https://user-images.githubusercontent.com/46431837/128733801-a9327250-a8ec-4e0b-a986-08d399a6d4b3.jpg)

## Assigning the VLAN to a VLANGroup, and assigning the VLANGroup to a site

- This works if the site is removed from the prefix.

URL for image: `https://netbox-url/ipam/prefixes/4/edit/`

![prefix-no-site-vlan-listed](https://user-images.githubusercontent.com/46431837/128733899-3b6bea6d-861e-4176-83c5-dac616f57e7e.jpg)

- If prefix is assigned back to a site, VLAN's are not listed even though prefix is assigned to the VLAN group.

URL for image: `https://netbox-url/ipam/prefixes/4/edit/`

![vlan-site-assigned-no-vlan-listed](https://user-images.githubusercontent.com/46431837/128734415-cf5d3479-bdd8-4584-90c6-37811d5e92c9.jpg)

## My intent

Below is a diagram mapping my intent. It is possible my ignorance of Netbox at this point is the problem.  My goal is to have `[VLAN-Group]<-->[VLAN-ID]<-->[Prefix]<-->[Site]` relationship.

![vlan-to-prefix-to-site](https://user-images.githubusercontent.com/46431837/128738862-d1116f52-b270-4064-8f93-ddfff9fa3107.png)

**Reply by sdktr** [reply]
In this example, is your L2 domain stretched between te two sites or site unique?

If the latter: netbox intent is to explicitly model the unique L2 broadcast domain (VLAN) and not tbr blueprint of it that replicates over sites. So you'd have to create two (site specific) VLANgroups with a copy of the vlans in each of them. 

I like your way of thinking though, where the network model is administrated once and applied to multiple implementations, in this case by using the prefix as glue.

**Reply by jmmey** [reply]
The L2 domain is not stretched between sites. These locations are separated by an SD-WAN overlay with multiple types of transport (_MPLS, Internet, LTE_). They are small retail branch offices with an identical L1/L2 architecture. 

The VLAN ID's used at the sites are always the same (10, 20, 30), but represent a different L3 subnet that fall under an assigned site IPv4 aggregate. Since these locations always share identical VLAN ID's, I attempted to leverage a common VLAN group. If in the future we need to provision say VLAN 40 at all sites, my hope was to add it to the single VLAN group. 

My thought was less data entry and less prone to human error. I will use a VLAN group per site with a copy of the VLAN in each as you suggest.

I really appreciate the help!
